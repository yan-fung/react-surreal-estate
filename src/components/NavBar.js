import React from "react";
import FacebookLogin from "react-facebook-login";
import { FaFacebookF } from "react-icons/fa";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import homeImage from "../styles/images/home.png";
import "../styles/navbar.css";

const NavBar = ({ onLogin, onLogout, userId }) => {
  return (
    <div className="navbar">
      <img src={homeImage} alt="home_image" />
      <ul className="navbar-links">
        <li className="narbar-links-item">
          <Link className="narbar-links__text" to="/">
            View Properties
          </Link>
        </li>
        <li className="narbar-links-item">
          <Link className="narbar-links__text" to="/add_property">
            Add Properties
          </Link>
        </li>
      </ul>
      {userId ? (
        <button className="sign-out-btn" type="button" onClick={onLogout}>
          Sign out
        </button>
      ) : (
        <div className="fb-login">
          <FacebookLogin
            appId="813549316783868"
            callback={onLogin}
            userId={userId}
            cssClass="my-facebook-button-class"
            icon={<FaFacebookF className="fb-icon" />}
          />
        </div>
      )}
    </div>
  );
};

export default NavBar;

NavBar.propTypes = {
  onLogin: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
  userId: PropTypes.string.isRequired,
};
