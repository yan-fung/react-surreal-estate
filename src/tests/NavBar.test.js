import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  it("renders navbar correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders View Properties correctly", () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    const text = screen.getByText("View Properties");

    expect(text).toBeInTheDocument();
  });
});
