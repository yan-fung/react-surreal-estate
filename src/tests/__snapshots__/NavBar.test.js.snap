// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders navbar correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="home_image"
      src="home.png"
    />
    <ul
      class="navbar-links"
    >
      <li
        class="narbar-links-item"
      >
        <a
          class="narbar-links__text"
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="narbar-links-item"
      >
        <a
          class="narbar-links__text"
          href="/add_property"
        >
          Add Properties
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
